drivers/elasticsearch/src/main/java/io/sbk/driver/Elasticsearch/Elasticsearch.java

```diff
@@ -0,0 +1,182 @@
+        config.password = params.getOptionValue("password", config.password);
+        config.url = params.getOptionValue("url", config.url);
+        config.index = params.getOptionValue("index", config.index);
+    }
+
+    @Override
+    public void openStorage(final ParameterOptions params) throws IOException {
+        try {
+            elasticsearchClient = connect();
+            Printer.log.info("ElasticSearch Client Connected.....");
+            String index1 = config.index.trim();
+            if (!indexExists(index1)) {
+                createIndex(elasticsearchClient, index1);
+            }
+        } catch (ElasticsearchException e ) {
+            Printer.log.error(e.getMessage());
+            throw  new RuntimeException(e);
+        }
+    }
+
+    private ElasticsearchClient connect() {
+        Printer.log.info("Attempting to connect to Elasticsearch...");
+        try {
+            final BasicCredentialsProvider credentialsProvider = new BasicCredentialsProvider();
+            credentialsProvider.setCredentials(AuthScope.ANY, new UsernamePasswordCredentials(config.user, config.password));
+
+            RestClientBuilder builder = RestClient.builder(HttpHost.create(config.url))
+                    .setHttpClientConfigCallback(httpClientBuilder -> httpClientBuilder.setDefaultCredentialsProvider(credentialsProvider));
+
+            RestClient restClient = builder.build();
+            ElasticsearchTransport transport = new RestClientTransport(restClient, new JacksonJsonpMapper());
+            return new ElasticsearchClient(transport);
+        } catch (ElasticsearchException e) {
+            Printer.log.error("Error connecting to Elasticsearch: " + e.getMessage());
+            throw new RuntimeException(e);
+        }
+    }
+
+    private  boolean indexExists(String indexName) {
+        try {
+            BooleanResponse response = elasticsearchClient.
+                    indices().exists(e -> e.index(indexName));
+            return response.value();
+        } catch (IOException e) {
+            e.printStackTrace();
+            return false;
+        }
+    }
+
+    private  void createIndex(ElasticsearchClient client, String indexName) {
+        try {
+            CreateIndexRequest createIndexRequest = CreateIndexRequest.of(c -> c.index(indexName));
+            CreateIndexResponse createIndexResponse = client.indices().create(createIndexRequest);
+
+            if (createIndexResponse.acknowledged()) {
+                Printer.log.info(indexName + "Created Successfully");
+            } else {
+                Printer.log.info("Index creation was not acknowledged.");
+            }
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
+
+    private void shutDownElasticsearch() {
+        //shut down for linux system (optional)
+        try {
+            String[] command = {"sh", "-c", "sudo systemctl stop elasticsearch"};
```

Comment:
this seems a problem .. it becomes the linux specific ; why do you stop entire elastic search ; its assumed the elastic serach may in in remote system too;